### PR TITLE
Fix/plot surf roi

### PR DESCRIPTION
**Pull request by juhuntenburg**
Fixing #1697 

The input roi_map can now be a) an array of the same length as the mesh, or b) a file that can be loaded with load_surf_data and returns an array of the same length as the mesh.

### CODE REVIEW

File: nilearn/plotting/surf_plotting.py [hunk 1]
```diff
@@ -503,35 +503,25 @@ def plot_surf_roi(surf_mesh, roi_map, bg_map=None,
...
-                         'vertices included in that ROI')
-    vmin, vmax = np.min(roi_map), 1 + np.max(roi_map)
-    display = plot_surf(surf_mesh, surf_map=roi_map, bg_map=bg_map,
+    roi = load_surf_data(roi_map)
+    vmin, vmax = np.min(roi), 1 + np.max(roi)
+
+    mesh = load_surf_mesh(surf_mesh)
+
+    if len(roi.shape) is not 1:
+        raise ValueError('roi_map can only have one dimension but has'
```
Comment (GaelVaroquaux) [inline nilearn/plotting/surf_plotting.py]: I think that you are missing a space at the end of the string

**Comment by GaelVaroquaux** [comment]
Maybe we should update the docstring on the input argument to be more clear about the input format.

File: nilearn/plotting/surf_plotting.py (hunk 1 above)
Comment (juhuntenburg) [reply nilearn/plotting/surf_plotting.py]: Done

**Comment by juhuntenburg** [comment]
> Maybe we should update the docstring on the input argument to be more clear about the input format.

As in, adding that it has to have the same length as the number of vertices on the supplied mesh?

File: doc/connectivity/connectome_extraction.rst [hunk 2]
```diff
@@ -105,8 +104,8 @@ of the estimator::
 .. topic:: **Exercise: computing sparse inverse covariance**
    :class: green
 
-   Compute and visualize a connectome on the first subject of the brain
-   development dataset downloaded with :func:`nilearn.datasets.fetch_development_fmri`
+   Compute and visualize a connectome on the first subject of the ADHD
+   dataset downloaded with :func:`nilearn.datasets.fetch_adhd`
```
Comment (GaelVaroquaux) [inline doc/connectivity/connectome_extraction.rst]: Hum, you have a broken merge :(.

Comment (juhuntenburg) [reply doc/connectivity/connectome_extraction.rst]: Ok, will fix it. Just out of interest, how come it doesn't show up as a merge conflict?

**Comment by juhuntenburg** [comment]
> If I look at the current docstring for roi_map, and I am not sure that it is still appropriate.

Can you specify? It says that the input can be an array or a file, which is how it works now. Happy to describe in more detail, just want to make sure I understand what exactly is your concern

**Comment by GaelVaroquaux** [comment]
> Can you specify? It says that the input can be an array or a file, which is how it works now.

Point taken. I read the diff wrong.

Here is what I would write: "a Numpy array. The value at each vertex is
an integer giving the label number."

How about that?

**Comment by KamalakerDadi** [comment]
Failures on the travis and appveyor should be fixed after merging this PR #1992 . Please don't worry about the failures.

**Comment by KamalakerDadi** [comment]
Failures on the travis and appveyor should be fixed after merging this PR #1992 . Please don't worry on the failures.

**Comment by GaelVaroquaux** [comment]
@juhuntenburg : can you merge master into this branch, to fix the failure.

**Comment by GaelVaroquaux** [comment]
OK, merging. Thank you!

**Pull request merged**